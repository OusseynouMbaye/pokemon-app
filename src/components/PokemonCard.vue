<template>
  <div class="poke-card">
    <div>
      <img :src="poke.image" alt="poke.title" />
      <h2>{{ poke.title }}</h2>
      <div class="poke-stats">
        <!-- <div v-for="type in poke.type" :key="type">
        <span>{{ type }}</span>
            </div> -->

        <div v-for="(value, key) in poke.stats" :key="key">
          <div class="flex flex-col">
            <span>{{ key }}: </span>
            <span>{{ value }}</span>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts" setup>
defineProps({
  poke: {
    type: Object,
    required: true,
  },
})
</script>

<style lang="css" scoped>
.poke-card {
  padding: 20px;
  width: 100%;
  cursor: pointer;
  border: 1px solid #39495c;
  border-radius: 10px;
  /* margin-bottom: 18px; */
}
.poke-card:hover {
  transform: scaleX(1.1) scaleY(1.1);
  box-shadow: 0px 8px 47px 0px rgba(0, 0, 0, 0.07);
  box-shadow: 0 3px 12px 0 rgba(0, 0, 0, 0.2);
  margin: 20px 0;
  transition: all 0.2s ease-in-out;
}
.poke-card img {
  width: 100%;
  height: 100%;
  object-fit: contain;
}
/* create a grid with 3 columns and responsive */
.poke-stats {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  padding: 0 10px;
}
</style>
