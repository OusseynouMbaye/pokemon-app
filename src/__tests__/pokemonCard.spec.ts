import PokemonCard from '@/components/PokemonCard.vue'
import { mount } from '@vue/test-utils'
import { describe, expect, it, test } from 'vitest'

describe('PokemonCard', () => {
  test('should render pokemon object', async () => {
    expect(PokemonCard).toBeTruthy()

    const wrapper = mount(PokemonCard, {
      props: {
        // objet
        poke: {
          id: 25,
          title: 'Pikachu',
          img: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/25.svg',
          stats: {
            hp: 35,
            attack: 55,
            defense: 40,
            'special-attack': 50,
            'special-defense': 50,
            speed: 90,
          },
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
