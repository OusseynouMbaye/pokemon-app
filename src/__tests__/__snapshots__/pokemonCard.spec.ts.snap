// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonCard > should render pokemon object 1`] = `
"<div data-v-63ebbf9b=\\"\\" class=\\"poke-card\\">
  <div data-v-63ebbf9b=\\"\\"><img data-v-63ebbf9b=\\"\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/25.svg\\" alt=\\"pokes.title\\">
    <h2 data-v-63ebbf9b=\\"\\">Pikachu</h2>
    <div data-v-63ebbf9b=\\"\\" class=\\"poke-stats\\">
      <!-- <div v-for=\\"type in poke.type\\" :key=\\"type\\">
        <span>{{ type }}</span>
            </div> -->
      <div data-v-63ebbf9b=\\"\\">
        <div data-v-63ebbf9b=\\"\\" class=\\"flex flex-col\\"><span data-v-63ebbf9b=\\"\\">hp: </span><span data-v-63ebbf9b=\\"\\">35</span></div>
      </div>
      <div data-v-63ebbf9b=\\"\\">
        <div data-v-63ebbf9b=\\"\\" class=\\"flex flex-col\\"><span data-v-63ebbf9b=\\"\\">attack: </span><span data-v-63ebbf9b=\\"\\">55</span></div>
      </div>
      <div data-v-63ebbf9b=\\"\\">
        <div data-v-63ebbf9b=\\"\\" class=\\"flex flex-col\\"><span data-v-63ebbf9b=\\"\\">defense: </span><span data-v-63ebbf9b=\\"\\">40</span></div>
      </div>
      <div data-v-63ebbf9b=\\"\\">
        <div data-v-63ebbf9b=\\"\\" class=\\"flex flex-col\\"><span data-v-63ebbf9b=\\"\\">special-attack: </span><span data-v-63ebbf9b=\\"\\">50</span></div>
      </div>
      <div data-v-63ebbf9b=\\"\\">
        <div data-v-63ebbf9b=\\"\\" class=\\"flex flex-col\\"><span data-v-63ebbf9b=\\"\\">special-defense: </span><span data-v-63ebbf9b=\\"\\">50</span></div>
      </div>
      <div data-v-63ebbf9b=\\"\\">
        <div data-v-63ebbf9b=\\"\\" class=\\"flex flex-col\\"><span data-v-63ebbf9b=\\"\\">speed: </span><span data-v-63ebbf9b=\\"\\">90</span></div>
      </div>
    </div>
  </div>
</div>"
`;
